src/main/scala/com/mozilla/telemetry/pings/package.scala

```diff
@@ -0,0 +1,225 @@
+                   `environment.addons`: Option[Addons],
+                   // Main ping fields preparsed by hindsight
+                   `payload.simpleMeasurements`: JValue,
+                   `payload.keyedHistograms`: JValue,
+                   `payload.histograms`: JValue,
+                   `payload.info`: JValue
+                 )
+
+  case class CrashPayload(
+                           crashDate: String,
+                           processType: Option[String],
+                           hasCrashEnvironment: Option[Boolean],
+                           metadata: Option[Map[String, String]],
+                           version: Option[Int]
+                         )
+
+  case class CrashPing(
+                        application: Application,
+                        clientId: Option[String],
+                        payload: CrashPayload,
+                        // Environment is omitted it's partially available under meta
+                        meta: Meta
+                      ){
+
+    def isMainCrash(): Boolean = {
+      payload.processType.getOrElse("main") == "main"
+    }
+
+    def timestamp(): Timestamp = {
+      new Timestamp(this.meta.Timestamp / 1000000)
+    }
+  }
+
+  case class PayloadInfo(subsessionLength: Option[Int])
+
+  case class MainPing(
+                       application: Application,
+                       clientId: Option[String],
+                       // Environment omitted because it's mostly available under meta
+                       meta: Meta
+                     ){
+    def getCountHistogramValue(histogram_name: String): Int = {
+      try {
+        this.meta.`payload.histograms` \ histogram_name \ "values" \ "0" match {
+          case JInt(count) => count.toInt
+          case _ => 0
+        }
+      } catch { case _: Throwable => 0 }
+    }
+
+    def getCountKeyedHistogramValue(histogram_name: String, key: String): Int = {
+      try {
+        this.meta.`payload.keyedHistograms` \ histogram_name \ key \ "values" \ "0" match {
+          case JInt(count) => count.toInt
+          case _ => 0
+        }
+      } catch { case _: Throwable => 0 }
+    }
+
+
+    def usageHours(): Float = {
+      try {
+        this.meta.`payload.info` \ "subsessionLength" match {
+          case JInt(length) => Math.min(25, Math.max(0, length.toFloat / 3600))
+          case _ => 0
+        }
+      } catch { case _: Throwable => 0 }
+    }
+
+    def timestamp(): Timestamp = {
+      new Timestamp(this.meta.Timestamp / 1000000)
+    }
+  }
+
+  case class Environment(build: EnvironmentBuild, system: EnvironmentSystem)
+
+  case class EnvironmentBuild(
+                               version: Option[String],
+                               buildId: Option[String],
+                               architecture: Option[String]
+                             )
+
+  case class EnvironmentSystem(os: OS)
+
+  case class OS(
+                 name: Option[String],
+                 version: Option[String])
```

Comment:
I reformatted these classes following the scala style guidelines. I also activated scalastyle, but unfortunately it doesn't provide any linting for this kind of problems.